### ISSUE TITLE
Generative pretraining code location 

### ISSUE THREAD

**Issue by olavdc** [bowang-lab/scGPT#230]
Hi @subercui, thanks for the cool project ! 

I'm trying to retrain the model from scratch with my own data. From earlier issues i've understood that the pretrain code lives on the `dev-temp` branch. I've been able to re-implement that and the model indeed seems to have a decreasing MSE loss. 

I'm nevertheless not understanding where the iterative part of the generative pretraining detailed in the `section 4.3.2 Attention mask for generative pre-training` actually takes place on the branch. As far as I understand the code the `generative_forward`call that happend in the `pretrain.py` script uses the forward function of this class: https://github.com/bowang-lab/scGPT/blob/dev-temp/scgpt/model/flash_layers.py#L389, but without engaging any iterative process.  

I also didn't observe the usage of the `generate` function of the model which seems to be a building block of the iterative generative process as it has an `gen_iters` argument.

Could you enlighten be on this point ? Thanks in advance !
